src/libraries/System.Net.Mail/src/System/Net/Mail/MailAddress.cs

```diff
@@ -63,6 +63,56 @@ public MailAddress(string address, string displayName) : this(address, displayNa
         //
         // A FormatException will be thrown if any of the components in 'address' are invalid.
         public MailAddress(string address, string displayName, Encoding displayNameEncoding)
+        {
+            bool parseSuccess = TryParse(address, displayName, displayNameEncoding, out (string displayName, string user, string host, Encoding displayNameEncoding) parsedData, throwExceptionIfFail: true);
+            _displayName = parsedData.displayName;
+            _userName = parsedData.user;
+            _host = parsedData.host;
+            _displayNameEncoding = parsedData.displayNameEncoding;
+
+            Debug.Assert(parseSuccess);
+        }
+
+        /// <summary>
```

Comment:
Thank you for adding the triple slash documentation for these new public APIs. We will port the comments to MS Docs when it's appropriate.